app/lib/account/backend.dart

```diff
@@ -180,6 +180,46 @@ class AccountBackend {
     return user;
   }
 
+  /// Returns [Like] if [user] likes [package], otherwise returns `null`.
+  Future<Like> getPackageLikeStatus(User user, String package) async {
+    final key =
+        _db.emptyKey.append(User, id: user.id).append(Like, id: package);
+    Like like;
+    try {
+      like = await _db.lookupValue(key);
+    } on KeyNotFoundException {
+      return null;
+    }
+    return like;
+  }
+
+  /// Return an iterable with the names of all the packages that the given [user] likes.
+  Future<List<String>> listPackageLikes(User user) async {
+    final likes = await _db.query<Like>(ancestorKey: user.key).run().toList();
+    return likes.map((Like l) => l.package).toList();
+  }
+
+  /// Creates an a package like entry for the given [user] and [package].
+  Future<void> likePackage(User user, String package) async {
```

Comment:
nit, returning the `Like` entity would be nice :)

That way we expose the `created` timestamp to the caller.